### PR TITLE
Feature/align cfg max error

### PR DESCRIPTION
**Pull request by rrahn**
Adds max_error config for edit distance.

Adds has_config helper to utility

Some minor changes.

### CODE REVIEW

**Comment by marehr** [comment]
A different failing test case:
```
In file included from /home/travis/build/seqan/seqan3/test/unit/core/tuple_utility_test.cpp:44:0:
/home/travis/build/seqan/seqan3/include/seqan3/core/pod_tuple.hpp: In instantiation of ‘struct seqan3::pod_tuple<bar, float>’:
/home/travis/build/seqan/seqan3/include/seqan3/core/pod_tuple.hpp:88:25:   recursively required from ‘struct seqan3::pod_tuple<long int, bar, float>’
/home/travis/build/seqan/seqan3/include/seqan3/core/pod_tuple.hpp:88:25:   required from ‘struct seqan3::pod_tuple<int, long int, bar, float>’
/home/travis/build/seqan/seqan3-build/vendor/googletest/googletest/include/gtest/internal/gtest-internal.h:636:42:   required from ‘std::vector<std::basic_string<char> > testing::internal::GenerateNames() [with NameGenerator = testing::internal::DefaultNameGenerator; Types = testing::internal::Types2<std::tuple<int, long int, bar, float>, seqan3::pod_tuple<int, long int, bar, float> >]’
/home/travis/build/seqan/seqan3/test/unit/core/tuple_utility_test.cpp:68:123:   required from here
/home/travis/build/seqan/seqan3/include/seqan3/core/pod_tuple.hpp:83:5: error: static assertion failed: If you are not going to insert a POD type, use std::tuple instead.
     static_assert(std::is_pod_v<type0>, SEQAN_NOT_POD);
     ^~~~~~~~~~~~~
```

File: include/seqan3/alignment/configuration/utility.hpp [hunk 1]
```diff
@@ -173,12 +175,13 @@ inline constexpr align_cfg::id align_config_type_to_id_v = align_config_type_to_
...
-      { true,  true,  false, true,  true  },  // score
-      { true,  true,  true,  false, true  },  // global
-      { true,  true,  true,  true,  false } } // output
+    //  gap    s_ends score  global output  m_err
+    { { false, true,  true,  true,  true , true },  // gap
+      { true,  false, true,  true,  true , true },  // sequence_ends
+      { true,  true,  false, true,  true , true },  // score
+      { true,  true,  true,  false, true , true },  // global
+      { true,  true,  true,  true,  false, true },  // output
+      { true,  true,  true,  true,  false, false} } // max_error
```
Comment (marehr) [inline include/seqan3/alignment/configuration/utility.hpp]: does that mean that `max_error` can't combined with `output`?

Comment (rrahn) [reply include/seqan3/alignment/configuration/utility.hpp]: only if output was set before max_error.
I find this kind of logical. 🤔

Comment (rrahn) [reply include/seqan3/alignment/configuration/utility.hpp]: Good spot 🤣

File: include/seqan3/core/algorithm/configuration.hpp [hunk 2]
```diff
@@ -143,11 +143,11 @@ class configuration : public std::tuple<configs_t...>
...
-    configuration(configuration const &)             = default;
-    configuration(configuration &&)                  = default;
-    configuration & operator=(configuration const &) = default;
-    configuration & operator=(configuration &&)      = default;
+    constexpr configuration()                                  = default;
+    constexpr configuration(configuration const &)             = default;
+    constexpr configuration(configuration &&)                  = default;
+    constexpr configuration & operator=(configuration const &) = default;
+    constexpr configuration & operator=(configuration &&)      = default;
     ~configuration()                                 = default;
```
Comment (marehr) [inline include/seqan3/core/algorithm/configuration.hpp]: what is with the destructor? indentation

File: include/seqan3/alignment/configuration/utility.hpp (hunk 1 above)
Comment (marehr) [reply include/seqan3/alignment/configuration/utility.hpp]: What do you mean by "good spot"?

**Comment by rrahn** [comment]
That is wrong.

File: include/seqan3/alignment/configuration/align_config_edit.hpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+#include <seqan3/alignment/configuration/align_config_score.hpp>
+#include <seqan3/alignment/scoring/all.hpp>
+
+namespace seqan3::align_cfg
+{
+
+//!\brief Shortcut for edit distance configuration.
+inline constexpr detail::configuration edit = global |
+                                              score(nucleotide_scoring_scheme{}) |
+                                              gap(gap_scheme{gap_score{1}});
```
Comment (joergi-w) [inline include/seqan3/alignment/configuration/align_config_edit.hpp]: This is not edit distance!
Use match_score{0} and mismatch_score{-1} for the scoring scheme and gap_score{-1} instead.

**Pull request merged**